lib/api/apiUtils/object/coldStorage.js

```diff
@@ -0,0 +1,107 @@
+const ObjectMDArchive = require('arsenal').models.ObjectMDArchive;
+const errors = require('arsenal').errors;
+const { config } = require('../../../Config');
+const { locationConstraints } = config;
+
+/**
+ * Check restore status.
+ *
+ * @param {ObjectMD} objectMD - object metadata
+ * @return {undefined}
+ */
+function _checkRestoreStatus(objectMD) {
```

Comment:
APIs with this kind of mixed return types are not very developer-friendly and are extremely error-prone (the fact that javascript allows it should not be seen as an encouragement).

It seems to me that this whole logic is just a yes/no question: can we initiate a new restore or not? There's no real need for an error (or even different variations of the status) when doing the computation, the error is just part of the response to the user.

Reply:
agree with you. changed to use one function to only check if we can initiate a restore(return error on failure, return undefined on success)